core-tests/shared/src/test/scala/zio/FiberFailureSpec.rs

```diff
@@ -0,0 +1,129 @@
+package zio
+
+import zio.test._
+import zio.test.Assertion._
+import java.io.{ByteArrayOutputStream, PrintStream}
+
+object FiberFailureSpec extends ZIOBaseSpec {
+  def spec = suite("FiberFailureSpec")(
+    test("FiberFailure getStackTrace includes relevant ZIO stack traces") {
+      val exception    = new Exception("Test Exception")
+      val fiberFailure = FiberFailure(Cause.fail(exception))
+      val stackTrace   = fiberFailure.getStackTrace
+
+      assertTrue(
+        stackTrace.exists(_.getClassName.contains("FiberFailureSpec")), // User code presence
+        stackTrace.exists(_.getClassName.startsWith("zio."))            // ZIO internals presence
+      )
+    },
+    test("FiberFailure toString should match cause.prettyPrint") {
+      val cause        = Cause.fail(new Exception("Test Exception"))
+      val fiberFailure = FiberFailure(cause)
+
+      assert(fiberFailure.toString)(equalTo(cause.prettyPrint))
```

Comment:
Why `cause.prettyPrint`? The `cause` might not include the stack traces

Reply:
You're right that `cause.prettyPrint` doesn't always include the full stack trace. The `toString` method of FiberFailure currently delegates to `cause.prettyPrint` which simplifies the output but might miss some essential details.

To ensure we're capturing the full context of the failure, including the complete stack trace, it might be better to adjust the toString method in FiberFailure to concatenate the pretty-printed cause with the full stack trace